### ISSUE TITLE
[Dev] Prompt whether an issue should be labeled as Breaking Change when merging

### ISSUE THREAD

**Issue by raulcd** [apache/arrow#38841]
### Describe the enhancement requested

Currently is rather difficult to know whether a commit contains a Breaking Change or not and this is done by manually tagging the issue with the `Breaking Change` label.
We added the PR template which prompts whether there are user-facing changes or not. I think we should modify our merge script to prompt whether the breaking change label has to be added on the issue when merging a PR.

Once this is done we could modify our Release Changelog to add a section of `Breaking changes` that could help users identify breaking changes for newer versions.

### Component(s)

Developer Tools

**Comment by amoeba** [comment]
This is great, thanks @raulcd. Are all commits on main made using the merge script or do some maintainers use other mechanisms such as the GitHub UI? I would want any prompt to apply everywhere if possible.

**Comment by raulcd** [comment]
All commits via PR are merged with the merge script. We do not use the GitHub UI. The only commits that are pushed directly to main are the bump versions after a release but those are not relevant for this.

**Comment by amoeba** [comment]
Great, thanks.

We might want another ticket for this idea but would it be possible to update out GitHub workflows so the breaking change label is copied from PR to issue? In the case of https://github.com/apache/arrow/pull/35656, the label was added to the PR but not the issue which seems like a totally reasonable thing to have done so it'd be nice if we could automate that problem away.

**Comment by thisisnic** [comment]
This issue has been marked as stale because it has had no activity in the past 2 years. Please remove the stale label or comment below, or this issue will be closed in 14 days.
